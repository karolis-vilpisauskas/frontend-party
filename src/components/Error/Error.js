import React from "react";
import "./Error.scss";

const Error = ({message}) => {
  return <div className={message ? "error is-active" : "error"}>{message}</div>;
};

export default Error;
