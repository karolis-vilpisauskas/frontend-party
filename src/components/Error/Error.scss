@import "styles/utils.scss";

.error {
  display: none ;
  max-width: 359px;
  height: auto;
  width: 80%;
  border-radius: 5px;
  margin: 10px auto;
  padding: 10px;
  background-color: $danger;
  text-align: center;
  font-size: 16px;
  font-weight: 300;
  line-height: 1.1;
  letter-spacing: 0.4px;
  color: $white;
  transition: all 0.5s linear;
  opacity: 0;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  &.is-active {
    display: block;
    animation: flash 4s linear;
  }
}

@keyframes flash {
  0% {
    opacity: 0;
  }
  20% {
    opacity: 1;
  }
  80% {
    opacity: 1;
  }
  100% {
    opacity: 0;
  }
}
