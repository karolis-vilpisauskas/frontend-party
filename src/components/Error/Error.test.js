import '@testing-library/jest-dom/extend-expect'
import React from "react";
import { cleanup, render } from "@testing-library/react";
import Error from "./Error";
afterEach(cleanup);

const createTestProps = props => ({
  ...props
});

const renderTest = () => {
  const props = createTestProps();
  const { getByTestId } = render(
    <Error {...props}>
    </Error>
  );
  const container = getByTestId("error");
  return {
    getByTestId,
    container
  };
};

describe("Error", () => {
  describe("rendering", () => {
    test("it renders without crashing", () => {
      const { container } = renderTest();
      expect(container).toMatchSnapshot();
    });
  });
});
