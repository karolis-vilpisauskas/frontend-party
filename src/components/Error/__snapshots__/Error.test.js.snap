// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error rendering it renders without crashing 1`] = `
<div
  class="error"
  data-testid="error"
/>
`;
